tests/opencascade/manifold_clone_01.cc

```diff
@@ -54,14 +54,16 @@ main()
     TopoDS_Edge edge           = BRepBuilderAPI_MakeEdge(circle);
 
     NormalProjectionBoundary<2, 3>  manifold_b(edge);
-    std::unique_ptr<Manifold<2, 3>> clone_b = manifold_b.clone();
+    std::unique_ptr<Manifold<2, 3>> clone_b       = manifold_b.clone();
+    const auto &                    deref_clone_b = *clone_b;
     deallog << "typeid of NormalProjectionBoundary<2, 3> is "
-            << boost::core::demangle(typeid(*clone_b).name()) << std::endl;
+            << boost::core::demangle(typeid(deref_clone_b).name()) << std::endl;
```

Comment:
What's the difference here?

Reply:
The compiler complains about a possible side-effect of dereferencing the `std::uniue_ptr` inside `typeid`:
```
8196: /mnt/data/darndt/Sources/dealii-clang-6.0.0/tests/opencascade/manifold_clone_01.cc:125:45: error: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Werror,-Wpotentially-evaluated-expression]
8196:             << boost::core::demangle(typeid(*clone_m).name()) << std::endl;
```

Reply:
Ah, how interesting!